core/conn/jdbc_type4/src/org/trafodion/jdbc/t4/HPT4Messages.java

```diff
@@ -251,18 +253,13 @@ static HPT4Exception createSQLException(T4Properties t4props, Locale msgLocale,
 			Object p[] = T4LoggingUtilities.makeParams(t4props, messageId, messageArguments);
 			log.logp(Level.SEVERE, "HPT4Messages", "createSQLException", "", p);
 		}
-
-		Locale currentLocale;
+		
+		Locale currentLocale = t4props == null ? null : t4props.getLocale();
+		currentLocale = currentLocale == null ? Locale.getDefault(): currentLocale;
+		
 		int sqlcode;
```

Comment:
Arvind, Dont worry about it. t4props is the only way to get locale and should be the only way. As i said, that is meaningless that some messages are Chinese but the others are English. If we do that, it will involve more misunderstandings.